### PR TITLE
[Fix][Relax][Frontend][Torch] Honor the `dtype` argument of `aten.mean` (`torch.Tensor.mean` / `torch.mean`)

### LINKED ISSUE #20230: [Bug][Relax][Frontend][PyTorch] `torch.Tensor.mean(..., dtype=...)` silently ignores the `dtype` argument and returns the input dtype
### Expected behavior

`torch.Tensor.mean` (and the equivalent `torch.mean`) accepts an optional keyword-only `dtype` argument that controls both the accumulation type and the output type:

```python
>>> x = torch.randn(2, 3, 4, dtype=torch.float32)
>>> x.mean(dim=1, dtype=torch.float64).dtype
torch.float64
```

When a model uses `mean(..., dtype=...)` with `dtype` different from the input dtype, the TVM PyTorch frontend should produce a tensor of the requested dtype, matching native PyTorch.

### Actual behavior

`tvm.relax.frontend.torch.from_exported_program` silently drops the `dtype` argument. The model imports, builds, and runs without any error, but the output tensor has the **input** dtype instead of the requested dtype. In the `fp16 → fp32` case the values are also computed at the wrong (lower) precision.

The converter `_mean` (`python/tvm/relax/frontend/torch/base_fx_graph_translator.py:1646`) only reads `dim` and `keepdim` and never reads `node.kwargs["dtype"]`:

```python
def _mean(self, node: fx.Node) -> relax.Var:
    args = self.retrieve_args(node)
    x = args[0]
    dim = args[1] if len(node.args) > 1 else node.kwargs.get("dim", None)
    keepdim = args[2] if len(node.args) > 2 else node.kwargs.get("keepdim", False)
    return self.block_builder.emit(relax.op.mean(x, dim, keepdims=keepdim))
```

`relax.op.mean` (`python/tvm/relax/op/statistical.py:54`) has no `out_dtype` parameter either, so there is no way the dtype is honored. The same converter is registered for `mean.dim` / `mean.default` (`exported_program_translator.py:1915-1916`) and for `mean` (`fx_translator.py:968`).

`torch.export` does preserve the `dtype` argument on the `aten.mean` node, so this is reachable from ordinary user code:

```
%mean : call_function[target=torch.ops.aten.mean.dim](args = (%x, [1]), kwargs = {dtype: torch.float64})
```

### Environment

- OS: Linux
- TVM: v0.24.dev0 (main branch, commit `262c6d2e0`; bug also present in the latest source at `390af87345`)
- Python: 3.11
- torch: 2.10.0+cu128

### Steps to reproduce

```python
"""Repro: torch.mean(dtype=...) output dtype dropped by TVM relax PyTorch frontend."""
import torch
import torch.nn as nn
import numpy as np
import tvm
from tvm import relax
from tvm.relax.frontend.torch import from_exported_program


class M(nn.Module):
    def forward(self, x):
        return x.mean(dim=1, dtype=torch.float64)


m = M().eval()
x = torch.randn(2, 3, 4, dtype=torch.float32)

with torch.no_grad():
    ref = m(x)                                   # float64, shape (2, 4)
print("torch ref :", ref.dtype, tuple(ref.shape))

exp = torch.export.export(m.cpu(), (x.cpu(),))
mod = from_exported_program(exp)
ex = relax.build(mod, target="llvm")
vm = relax.VirtualMachine(ex, tvm.cpu())
out = vm["main"](x.numpy())[0].numpy()
print("tvm out   :", out.dtype, out.shape)       # float32, shape (2, 4)  <-- wrong dtype
```

Actual output:

```
torch ref : torch.float64 (2, 4)
tvm out   : float32 (2, 4)
```

The full differential matrix (`prove_hum/torch_mean/1复现_torch_mean.py`) shows 8/8 baseline cases (global / single / multi-dim / keepdim / negative axis / same-dtype control) matching native PyTorch exactly, and 6/6 `dtype != input dtype` cases diverging in output dtype:

| input | call | torch output | TVM output |
|---|---|---|---|
| fp32 | `mean(dim=1, dtype=fp64)` | `float64` | `float32` |
| fp32 | `mean(dtype=fp64)` | `float64` | `float32` |
| fp32 | `mean(dim=(1,2), keepdim=True, dtype=fp64)` | `float64` | `float32` |
| fp64 | `mean(dim=1, dtype=fp32)` | `float32` | `float64` |
| fp16 | `mean(dim=1, dtype=fp32)` | `float32` | `float16` |
| fp32 | `mean(dim=1, dtype=fp16)` | `float16` | `float32` |

### Triage

* needs-triage
* bug
* relax
* frontend/torch

### PR DESCRIPTION
**Pull request by siyiweigeHEW**
Fixes: #20230

## Summary

PyTorch's `Tensor.mean` (and the equivalent `torch.mean`) accepts an
optional keyword-only `dtype` argument that controls **both** the
accumulation type and the output type. `torch.export` preserves it on the
`aten.mean.dim` / `aten.mean.default` nodes, but the `_mean` converter in
`base_fx_graph_translator.py` never reads `node.kwargs["dtype"]`, so the
argument is silently dropped and the output keeps the input dtype. For
example, `x.mean(dim=1, dtype=torch.float64)` on an fp32 `x` returns
`float32` instead of `float64`, and in the fp16→fp32 direction the values
are also accumulated at the wrong (lower) precision.

This PR makes `_mean` honor `dtype` by casting the input to the requested
dtype before reducing — the same pattern `_sum` already uses.

## Root cause

`_mean` (`python/tvm/relax/frontend/torch/base_fx_graph_translator.py:1646`)
only reads `dim` and `keepdim`:

```python
def _mean(self, node: fx.Node) -> relax.Var:
    args = self.retrieve_args(node)
    x = args[0]
    dim = args[1] if len(node.args) > 1 else node.kwargs.get("dim", None)
    keepdim = args[2] if len(node.args) > 2 else node.kwargs.get("keepdim", False)
    return self.block_builder.emit(relax.op.mean(x, dim, keepdims=keepdim))
```

`relax.op.mean` (`python/tvm/relax/op/statistical.py:54`) has no
`out_dtype` parameter, so the requested dtype cannot be propagated.
`torch.export` does preserve the `dtype` kwarg on the node:

```
%mean : call_function[target=torch.ops.aten.mean.dim](args = (%x, [1]), kwargs = {dtype: torch.float64})
```

The same converter is registered for `mean.dim` / `mean.default`
(exported-program path) and for `mean` (fx-trace path), so both entry
points are affected.

## Fix

When `dtype` is present on the node, cast the input to the requested dtype
before emitting the mean, mirroring the existing `_sum` handling:

```python
dtype = node.kwargs.get("dtype", None)
if dtype is not None:
    x = self.block_builder.emit(
        relax.op.astype(x, self._convert_data_type(dtype, self.env))
    )
return self.block_builder.emit(relax.op.mean(x, dim, keepdims=keepdim))
```

Casting the input and then reducing is exactly PyTorch's documented
semantics: the `dtype` argument controls the accumulation type (the input
is converted to `dtype` before the reduction), so no precision is lost
between the cast and the mean.

### In-tree regression test

Extended `test_mean` in `tests/python/relax/test_frontend_from_fx.py` with
`MeanDtype`, an fx-traced module calling `input.mean(-1,
dtype=torch.float64)`, and its expected IR: an `R.astype` to `"float64"`
followed by `R.mean` producing a `float64` output. The produced module is
checked with `tvm.ir.assert_structural_equal` via the file's `verify_model`
helper.

### Differential test

The prove_hum differential harness (`1复现_torch_mean.py`, converting
`torch.export` output with `from_exported_program` and comparing against
native PyTorch 2.10.0) was run on a runnable build of this change — 15
cases:

- **Baseline (7)**: global / single-dim / multi-dim / keepdim /
  negative-axis `mean` without `dtype` — unchanged, all match native
  PyTorch (dtype, shape, values).
- **Same-dtype control (1)**: `mean(dtype=fp32)` on fp32 — matches.
- **`dtype != input` (6)**: `fp32→fp64`, `fp64→fp32`, `fp16→fp32`,
  `fp32→fp16`, including global and multi-dim + keepdim variants.

All 6 `dtype != input` cases now produce the requested output dtype in
every direction (pre-fix, all 6 returned the input dtype). Values match to
native PyTorch within fp16/fp32 rounding precision. Both converter entry
paths — `from_exported_program` and `from_fx` — were verified.

## Files changed

- `python/tvm/relax/frontend/torch/base_fx_graph_translator.py` — `_mean`
  reads `node.kwargs["dtype"]` and casts the input with
  `relax.op.astype` before emitting `relax.op.mean`.
- `tests/python/relax/test_frontend_from_fx.py` — add `MeanDtype` /
  `ExpectedDtype` coverage to `test_mean`.

### CODE REVIEW

**Review by tlopex — APPROVED** [review]
LGTM
